### PR TITLE
sci_tactical_plugin port to ROS2

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

This PR includes updates to port the sci_tactical_plugin from ROS1 to ROS2.

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/1906

## How Has This Been Tested?

All unit tests pass, and local integration tests have been conducted.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: stop_controlled_intersection_tactical_plugin/include/stop_controlled_intersection_plugin.hpp [hunk 1]
```diff
@@ -157,30 +138,38 @@ class StopControlledIntersectionTacticalPlugin
...
 
   carma_wm::WorldModelConstPtr wm_;
   StopControlledIntersectionTacticalPluginConfig config_;
-  PublishPluginDiscoveryCB plugin_discovery_publisher_;
- 
-  cav_msgs::Plugin plugin_discovery_msg_;
-  carma_debug_msgs::TrajectoryCurvatureSpeeds debug_msg_;
+
+  // Unit test helper functions
+  carma_wm::WorldModelConstPtr get_wm() { return wm_; }
```
Comment (adev4a) [inline stop_controlled_intersection_tactical_plugin/include/stop_controlled_intersection_plugin.hpp:161]: Why do we need these? Can't we just access the wm_ pointer?

File: stop_controlled_intersection_tactical_plugin/launch/stop_controlled_intersection_tactical_plugin.launch.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        package='carma_ros2_utils',
+        name='stop_controlled_intersection_tactical_plugin_container',
+        namespace=GetCurrentNamespace(),
+        executable='carma_component_container_mt',
+        composable_node_descriptions=[
+            
+            # Launch the core node(s)
+            ComposableNode(
+                package='stop_controlled_intersection_tactical_plugin',
+                plugin='stop_controlled_intersection_tactical_plugin::SCIStrategicPlugin',
```
Comment (adev4a) [inline stop_controlled_intersection_tactical_plugin/launch/stop_controlled_intersection_tactical_plugin.launch.py:52]: This node should be stop_controlled_intersection_tactical_plugin::StopControlledIntersectionTacticalPlugin

Comment (MishkaMN) [reply stop_controlled_intersection_tactical_plugin/launch/stop_controlled_intersection_tactical_plugin.launch.py]: changed

File: stop_controlled_intersection_tactical_plugin/include/stop_controlled_intersection_plugin.hpp (hunk 1 above)
Comment (MishkaMN) [reply stop_controlled_intersection_tactical_plugin/include/stop_controlled_intersection_plugin.hpp]: somehow I wasn't able to access wm_ before even if I added FRIEND_TEST, but it does work now. Maybe I had a typo. I removed these helper functions now.

**Review by adev4a — APPROVED** [review]

**Pull request merged**